src/opnsense/scripts/dhcp/get_leases6.py

```diff
@@ -0,0 +1,186 @@
+
+        if field_value is not None:
+            segment[field_name] = field_value
+
+    return (seg_type, segment)
+
+def parse_iaid_duid(input):
+    """
+    parse the combined IAID_DUID value. This is provided in the form
+    of ascii characters. Non-printable characters are provided as octal escapes.
+    We return the hex representation of the raw IAID_DUID value, the IAID integer,
+    as well as the separated DUID value in a three-tuple. The IAID_DUID value is
+    used to uniquely identify a lease, so this value should be used to determine the last
+    relevant entry in the leases file.
+    """
+    input = input[1:-1] # strip double quotes
+    parsed = []
+    i = 0
+    while i < len(input):
+        c = input[i]
+        if c == '\\':
+            next_c = input[i + 1]
+            if next_c == '\\' or next == '"':
+                parsed.append("%02x" % ord(next_c))
+                i += 1
+            elif next_c.isnumeric():
+                octal_to_decimal = int(input[i+1:i+4], 8)
+                parsed.append("%02x" % octal_to_decimal)
+                i += 3
+        else:
+            parsed.append("%02x" % ord(c))
+        i += 1
+
+    iaid = int(''.join(reversed(parsed[0:4])))
+    duid = ":".join([str(a) for a in parsed[4:]])
+    iaid_duid = ":".join([str(a) for a in parsed])
+    return (iaid_duid, iaid, duid)
+
+def parse_lease(lines):
+    """
+    Parse a DHCPv6 lease. We return a two-tuple containing the combined iaid_duid
+    and the lease. a single lease may contain multiple addresses/prefixes.
+    """
+    lease = dict()
+    cur_segment = []
+    addresses = []
+    prefixes = []
+    (iaid_duid, iaid, duid) = parse_iaid_duid(lines[0].split()[1])
+    lease['lease_type'] = lines[0].split()[0]
+    lease['iaid_duid'] = iaid_duid
+    lease['iaid'] = iaid
+    lease['duid'] = duid
+
+    for line in lines:
+        parts = line.split()
+
+        cltt = None
+        if parts[0] == 'cltt' and len(parts) >= 3:
+            cltt = parse_date(parts[2], parts[3])
+        if cltt is not None:
+            lease['cltt'] = cltt
+
+        if parts[0] == 'iaaddr' or parts[0] == 'iaprefix':
+            cur_segment.append(line)
+        elif len(line) > 3 and line[2] == '}' and len(cur_segment) > 0:
+            cur_segment.append(line)
+            (segment_type, segment) = parse_iaaddr_iaprefix(cur_segment)
+            if segment_type == 'iaaddr':
+                addresses.append(segment)
+            if segment_type == 'iaprefix':
+                prefixes.append(segment)
+            cur_segment = []
+        elif len(cur_segment) > 0:
+            cur_segment.append(line)
+
+    # ia_ta/ia_na (addresses) and ia_pd (prefixes) are mutually exclusive.
+    if addresses:
+        lease['addresses'] = addresses
+    if prefixes:
+        lease['prefixes'] = prefixes
+
+    return (iaid_duid, lease)
+
+if __name__ == '__main__':
+    app_params = {'inactive': '0'}
+    params.update_params(app_params)
```

Comment:
I would opt for `argparse` here, I know I added the "params" a long time ago, but argparse is cleaner and removes the need for the custom `sys.path.insert(0, "/usr/local/opnsense/site-python")`

Reply:
i agree, this was quick & dirty - but argparse prefers boolean interpretations as either '--value' or nothing, which doesn't really translate well to configd action parameters. I'll update it to a string argument in argparse.